packages/package-manager/src/PackageManager.ts

```diff
@@ -17,14 +18,20 @@ export function getPossibleProjectRoot(): string {
   return realpathSync(process.cwd());
 }
 
-export function spawnSudoAsync(command: string): Promise<void> {
-  const packageJSON = require('../package.json');
-  return new Promise((resolve, reject) => {
-    sudo.exec(command, { name: packageJSON.name }, error => {
-      if (error) {
-        reject(error);
-      }
-      resolve();
+export async function spawnSudoAsync(command: string[], spawnOptions: SpawnOptions): Promise<void> {
+  // sudo prompt only seems to work on win32 machines.
+  if (process.platform === 'win32') {
+    return new Promise((resolve, reject) => {
+      sudo.exec(command.join(' '), { name: 'pod install' }, error => {
+        if (error) {
+          reject(error);
+        }
+        resolve();
+      });
     });
-  });
+  } else {
+    // Attempt to use sudo to run the command on Mac and Linux.
+    // TODO(Bacon): Make a v of sudo-prompt that's win32 only for better bundle size.
+    await spawnAsync('sudo', command, spawnOptions);
```

Comment:
we should show some message to the user to let them know why we are asking them to type in their password